frontend/src/pages/AdminPanel.js

```diff
@@ -0,0 +1,1253 @@
+      confirmPassword: selectedUser.confirmPassword,
+      isAdmin: Boolean(selectedUser.isAdmin),
+    };
+  
+    if (updatedUser.newName) {
+      axiosClient
+        .put(`/changeName/${selectedUser.id}`, {
+          newName: updatedUser.newName,
+        })
+        .then(() => {
+          setUsers((prevUsers) =>
+            prevUsers.map((user) =>
+              user.id === selectedUser.id ? { ...user, name: updatedUser.newName } : user
+            )
+          );
+        })
+        .catch((error) => {
+          console.error('Error updating name:', error);
+        });
+    }
+  
+    if (updatedUser.newEmail) {
+      axiosClient
+        .put(`/changeEmail/${selectedUser.id}`, {
+          newEmail: updatedUser.newEmail,
+        })
+        .then(() => {
+          setUsers((prevUsers) =>
+            prevUsers.map((user) =>
+              user.id === selectedUser.id ? { ...user, email: updatedUser.newEmail } : user
+            )
+          );
+        })
+        .catch((error) => {
+          console.error('Error updating email:', error);
+        });
+    }
+  
+    if (updatedUser.newPassword && updatedUser.confirmPassword) {
+      axiosClient
+        .put(`/changePassword/${selectedUser.id}`, {
+          newPassword: updatedUser.newPassword,
+          confirmPassword: updatedUser.confirmPassword,
+        })
+        .then(() => {
+          console.log('Password updated successfully');
+        })
+        .catch((error) => {
+          console.error('Error updating password:', error);
+        });
+    }
+  
+    if (updatedUser.isAdmin !== undefined) {
+      axiosClient
+        .put(`/changeUserRole/${selectedUser.id}`, {
+          isAdmin: updatedUser.isAdmin,
+        })
+        .then(() => {
+          setUsers((prevUsers) =>
+            prevUsers.map((user) =>
+              user.id === selectedUser.id ? { ...user, isAdmin: updatedUser.isAdmin } : user
+            )
+          );
+        })
+        .catch((error) => {
+          console.error('Error updating role:', error.response?.data || error.message);
+        });
+    }    
+    closeModal();
+  };
+
+  const openModal = (user) => {
+    setSelectedUser({ ...user });
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setSelectedUser(null);
+    setIsModalOpen(false);
+  };
+
+  const openDeleteConfirm = (user) => {
+    setSelectedUser(user);
+    setIsDeleteConfirmOpen(true);
+  };
+
+  const closeDeleteConfirm = () => {
+    setSelectedUser(null);
+    setIsDeleteConfirmOpen(false);
+  };
+
+  const deleteUser = () => {
+    axiosClient
+      .delete(`/deleteUser/${selectedUser.id}`)
+      .then(() => {
+        setUsers((prevUsers) => prevUsers.filter((user) => user.id !== selectedUser.id));
+        closeDeleteConfirm();
+      })
+      .catch((error) => {
+        console.error('Error deleting user:', error);
+      });
+  };  
+
+  {/* Kebabs API */}
+  const [kebabs, setKebabs] = useState([]);
```

Comment:
Same with kebabs, there's only an option to edit them, but we need to add them too